openhands/server/session/agent_session.py

```diff
@@ -160,6 +170,38 @@ def _create_security_analyzer(self, security_analyzer: str | None):
                 security_analyzer, SecurityAnalyzer
             )(self.event_stream)
 
+    def _clone_repo(self, github_token: str | None, selected_repository: str | None):
```

Comment:
you'll probably want to put these two new functions somewhere else, e.g. on `openhands/runtime/base.py`, so we can reuse them in other contexts